### DISCUSSION TITLE
useHydrateAtoms not updating after revalidatePath() in Next.js

### DISCUSSION THREAD

**Discussion by eckoln** [pmndrs/jotai#2087] [Q&A]
Hello. I am trying to make a list in `Next.js v13.4.19`.
In practice, I manage to set data I passed from `server component` to `client component` with `useHydrateAtoms` at first load. But, when I do `revalidatePath()` using in `Server Action`, useHydrateAtoms does not work although the updated data comes to client component.

Here is my sample code:
```js
// server component
export async function FollowPanel(){
  const followlist = await getUserFollowlist();
  return (
    ...
    <FollowlistAddItem followlist={followlist} />
  )
}

// client component
"use client"

import { atom, useAtom } from "jotai";
import { useHydrateAtoms } from "jotai/utils";

const followlistStore= atom([]);

export function FollowlistAddItem({ followlist }) {
  useHydrateAtoms([[followlistStore, followlist]]);
  const [followlist, setFollowlist] = useAtom(followlistStore);

  return (...)
}
  ```
  
After revalidatePath(), how can I set updated data?

**Comment by dai-shi** [comment]
Assuming the client component rerenders on revalidatePath, you need to use useEffect to change atom values for subsequent updates.
```js
export function FollowlistAddItem({ followlist }) {
  useHydrateAtoms([[followlistStore, followlist]]);
  const [currFollowlist, setFollowlist] = useAtom(followlistStore);
  useEffect(() => setFollowList(followList), [setFollowList, followList]);

  return (...)
}
```

**Reply by eckoln** [reply]
Thank you, this answer solved my issue.
